content/en/tracing/guide/add_span_md_and_graph_it.md

```diff
@@ -0,0 +1,287 @@
+
+    **Note**: take note of the service and [resource names][2] you are working on, these will come in handy later. In this example the service is the Ruby server `web-store` and the resource (endpoint) is `ShoppingCartController#checkout`.
+
+{{< tabs >}}
+{{% tab "Java" %}}
+
+The Datadog UI uses tags to set span level metadata. A full list of these metadata can be found in the [Datadog][1] and [OpenTracing][2] APIs.
+
+Custom metadata may be set for auto-instrumentation by grabbing the active span out of the global tracer and setting a tag with `setTag`.
+
+```java
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet
+class ServletImpl extends AbstractHttpServlet {
+  @Override
+  void doGet(HttpServletRequest req, HttpServletResponse resp) {
+    final Span span = GlobalTracer.get().activeSpan();
+    if (span != null) {
+      span.setTag("customer.id", 12345);
+      span.setTag("http.url", "/login");
+    }
+    // servlet impl
+  }
+}
+```
+
+[1]: https://github.com/DataDog/dd-trace-java/blob/master/dd-trace-api/src/main/java/datadog/trace/api/DDTags.java
+[2]: https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/tag/Tags.java
+{{% /tab %}}
+{{% tab "Python" %}}
+
+Add metadata directly to a span by calling `set_tag`. For example, with the following route handler:
+
+```python
+from ddtrace import tracer
+
+@app.route('/customer/<int:customer_id>')
+def handle_customer(customer_id):
+  with tracer.trace('web.request') as span:
+    span.set_tag('customer.id', customer_id)
+```
+
+The current span can be retrieved from the context in order to set its metadata. This way, if a span was started by the instrumentation, you can retrieve the span and add custom metadata. **Note**: If a span does not exist, `None` is returned:
+
+```python
+from ddtrace import tracer
+
+@app.route('/customer/<int:customer_id>')
+@tracer.wrap()
+def handle_customer(customer_id):
+  # get the active span in the context, put there by tracer.wrap()
+  current_span = tracer.current_span()
+  if current_span:
+    current_span.set_tag('customer.id', customer_id)
+```
+
+{{% /tab %}}
+{{% tab "Ruby" %}}
+
+Add metadata directly to `Datadog::Span` objects by calling `#set_tag`:
+
+```ruby
+# An example of a Sinatra endpoint,
+# with Datadog tracing around the request.
+get '/posts' do
+  Datadog.tracer.trace('web.request') do |span|
+    span.set_tag('http.url', request.path)
+  end
+end
+```
+
+Access the current active span from any method within your code. **Note**: If the method is called and there is no span currently active, `active_span` is `nil`.
+
+```ruby
+# e.g. adding tag to active span
+
+current_span = Datadog.tracer.active_span
+current_span.set_tag('<TAG_KEY>', '<TAG_VALUE>') unless current_span.nil?
```

Comment:
Let's not use a placeholder. Let's use the real example that is `customer_id` as we wrote for Java.

Also you can copy the previous example such as:
```ruby
# An example of a Sinatra endpoint,
# with Datadog auto-instrumentation enabled.
get '/posts' do
  current_span = Datadog.tracer.active_span
  current_span.set_tag('customer_id', 12345) unless current_span.nil?
end
```

Side note, we can make it parametrized as we did for the Python section to avoid using `12345` that is not really meaningful as example. cc @delner that can help with a better example.

Reply:
Sorry, addressed this in the previous comment. I'll leave with `12345` unless @delner has a better example.

Reply:
Parameter as in a URL parameter from Sinatra? I don't think it'd add much, but if you really think that's better I can figure something out.